nominal/cli/util/global_decorators.py

```diff
@@ -0,0 +1,122 @@
+        if verbose == 0:
+            log_level = logging.WARNING
+        elif verbose == 1:
+            log_level = logging.INFO
+        elif verbose >= 2:
+            log_level = logging.DEBUG
+
+        install_log_handler(level=log_level, no_color=no_color)
+        return func(*args, **kwargs)
+
+    return color_option(verbosity_option(wrapped_function))
+
+
+def debug_switch(func: typing.Callable[Param, T]) -> typing.Callable[..., T]:
+    """Decorator to add click options to a click command for dynamically spawning a post-mortem debugger at the
+    site of any raised exceptions that would have otherwise crashed the program.
+
+    This is primarily directed at developers, and as such, the --debug flag that is added to click commands
+    is witheld from the emitted --help text.
+    """
+    debug_option = click.option(
+        "--debug",
+        is_flag=True,
+        hidden=True,
+        help="Spawn a python debugger upon any exception being raised",
+    )
+
+    @functools.wraps(func)
+    def wrapped_function(*args: Param.args, debug: bool, **kwargs: Param.kwargs) -> T:
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            if debug:
+                pdb.post_mortem()
+
+            raise e
+
+    return debug_option(wrapped_function)
+
+
+def global_options(func: typing.Callable[Param, T]) -> typing.Callable[..., T]:
+    """Helper decorator that combines together debug_switch and verbosity_switch to provide commonly and globally
+    used utility options across all CLI endpoints within the nominal python client.
+    """
+    return functools.wraps(func)(debug_switch(verbosity_switch(func)))
+
+
+def client_options(func: typing.Callable[Param, T]) -> typing.Callable[..., T]:
+    """Decorator to add click options to a click command for dynamically creating and injecting an instance of the
+    NominalClient into commands based on user-provided flags containing the base API url and a path to a configuration
+    file containing an API Access Key.
+
+    This will add two options, --base-url and --token, which perform the two aforementioned configurations before spawning
+    a NominalClient.
+
+    NOTE: any click command utilizing this decorator MUST accept a key-value argument pair named client of type NominalClient.
+    """
+    url_option = click.option(
+        "--base-url",
+        default="https://api.gov.nominal.io/api",
+        show_default=True,
+        help="Base URL of the Nominal API to hit. Useful for hitting other clusters, e.g., staging for internal users.",
+    )
+    token_option = click.option(
+        "--token",
+        default=pathlib.Path("~/.nominal.yml").expanduser().resolve(),
+        type=click.Path(dir_okay=False, exists=True, resolve_path=True, path_type=pathlib.Path),
+        show_default=True,
+        help="Path to the yaml file containing the Nominal access token for authenticating with the API",
+    )
```

Comment:
this is semantically different than what existed previously right? I'd rather allow explicitly passing tokens in - this is 50% for debugging, so prefer the parameterization to be explicit rather than indirect